src/sync/incomingSyncUtils.js

```diff
@@ -43,11 +86,91 @@ export function integrateRecord(database, settings, syncRecord) {
       if (!syncRecord.RecordID) return; // If missing record id, ignore
       deleteRecord(database, internalRecordType, syncRecord.RecordID);
       break;
+    case 'merge':
+      mergeRecords(database, settings, internalRecordType, syncRecord);
     default:
-      return; // Silently ignore change types we don't handle, e.g. merges
+      return;
   }
 }
 
+/**
+ * Merge two existing records. One record is retained and the other is merged. After the objects
+ * are merged, the merged object is deleted from the database. The merge operation updates the
+ * fields of the retained record to reference the same data as the merged object.
+ *
+ * @param {Realm}  database           The local database
+ * @param {object} settings           Access to app settings
+ * @param {string} internalRecordType Internal record type for merge operation
+ * @param {object} syncRecord         Data representing the sync record
+ */
+export function mergeRecords(database, settings, internalRecordType, syncRecord) {
+  const recordToKeep = database
+    .objects(internalRecordType)
+    .filtered('id == $0', syncRecord.mergeIDtokeep)[0];
+  const recordToMerge = database
+    .objects(internalRecordType)
+    .filtered('id == $0', syncRecord.mergeIDtodelete)[0];
+
+  const recordsExist = recordToKeep && recordToMerge;
+  if (!recordsExist) return;
+
+  const tablesToUpdate = RECORD_TYPE_TO_TABLE[internalRecordType];
+  if (!tablesToUpdate) return; // TODO: log to bugsnag if merging not implemented for a certain recordType.
+
+  Object.entries(tablesToUpdate).forEach(
+    ([tableToUpdate, { field: fieldToUpdate, setter: fieldSetter }]) => {
+      const recordsToUpdate = database
+        .objects(tableToUpdate)
+        .filtered(`${fieldToUpdate}.id == $0`, recordToMerge.id)
+        .snapshot();
+      recordsToUpdate.forEach(record => {
+        if (record) {
+          // TODO: automatically add Transaction to otherParty.transactions when Transaction.otherParty is set
```

Comment:
I think this TODO has been fulfilled by setOtherParty on Transaction

Reply:
Removed.